File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/character/creation/CharacterCreationResultMessage.scala





















// Generated on 05/03/2013 10:34:37
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait CharacterCreationResultMessage extends MessageDeserializer {
  val messageId = 161

  protected trait Message extends BaseMessage with Serializable {
    val messageId = CharacterCreationResultMessage.messageId

    val result: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(result)
    }

  }

  protected def create(result: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val result = buf.readByte()

    create(result)
  }

}

object CharacterCreationResultMessage extends CharacterCreationResultMessage {
  case class MessageImpl(result: Byte) extends Message
  type Target = MessageImpl

  override def create(result: Byte) = MessageImpl(result)

  def apply(result: Byte) = create(result)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/CharacterCreationResultMessage.java


// Generated on 05/08/2013 19:37:41
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CharacterCreationResultMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 161;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte result;
    
    public CharacterCreationResultMessage() { }
    
    public CharacterCreationResultMessage(byte result) {
        this.result = result;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(result);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        result = buf.readByte();
        if (result < 0)
            throw new RuntimeException("Forbidden value on result = " + result + ", it doesn't respect the following condition : result < 0");
    }
    
}
